packages/cli/src/commands/hydrogen/link.ts

```diff
@@ -145,10 +145,20 @@ export async function linkStorefront(
 
 async function createNewStorefront(root: string, session: AdminSession) {
   const projectDirectory = basename(root);
+  const storefronts = await getStorefronts(session);
```

Comment:
Instead of calling `getStorefronts` here again, which will probably fire a new request and await it, can you pass the `storefronts` list from the caller as an argument directly? I don't think we have cache in place to dedup requests 😅